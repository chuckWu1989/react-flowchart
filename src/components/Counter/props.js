import PropTypes from 'prop-types';

export const defaultProps = {
  count: 0,
};
export const propTypes = {
  id: PropTypes.string.isRequired,
  count: PropTypes.number,
  willMount: PropTypes.func,
  increase: PropTypes.func,
  decrease: PropTypes.func,
  onDispose: PropTypes.func,
};
