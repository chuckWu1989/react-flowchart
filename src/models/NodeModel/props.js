export const TYPE = 'type';
export const defaultProps = {
  [TYPE]: 'progress',
};
