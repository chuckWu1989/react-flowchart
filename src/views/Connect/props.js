import PropTypes from 'prop-types';

export const TITLE = 'title';
export const SIZE = 'size';
export const BKCOLOR = 'bkColor';
export const STYLE = 'style';
export const TITLESTYLE = 'titleStyle';
export const defaultProps = {
  [TITLE]: 'Y',
  [SIZE]: 50,
  [BKCOLOR]: '#007b43',
  [STYLE]: {},
  [TITLESTYLE]: {},
};
export const propTypes = {
  [TITLE]: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.node]),
  [SIZE]: PropTypes.number,
  [BKCOLOR]: PropTypes.string,
  [STYLE]: PropTypes.object,
  [TITLESTYLE]: PropTypes.object,
};
