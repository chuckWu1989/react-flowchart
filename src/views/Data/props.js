import PropTypes from 'prop-types';

export const WIDTH = 'width';
export const HEIGHT = 'height';
export const STYLE = 'style';
export const defaultProps = {
  [WIDTH]: 150
};
