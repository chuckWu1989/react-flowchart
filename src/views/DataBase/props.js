import PropTypes from 'prop-types';

export const TITLE = 'title';
export const CONTENT = 'content';
export const WIDTH = 'width';
export const HEIGHT = 'height';
export const BKCOLOR = 'bkColor';
export const STYLE = 'style';
export const TITLESTYLE = 'titleStyle';
export const CONTENTSTYLE = 'contentStyle';
export const defaultProps = {
  [TITLE]: 'Untitled',
  [CONTENT]: undefined,
  [WIDTH]: 80,
  [HEIGHT]: 100,
  [BKCOLOR]: '#007bbb',
  [STYLE]: {},
  [TITLESTYLE]: {},
  [CONTENTSTYLE]: {},
};
export const propTypes = {
  [TITLE]: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.node]),
  [CONTENT]: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.node]),
  [WIDTH]: PropTypes.number,
  [HEIGHT]: PropTypes.number,
  [BKCOLOR]: PropTypes.string,
  [STYLE]: PropTypes.object,
  [TITLESTYLE]: PropTypes.object,
  [CONTENTSTYLE]: PropTypes.object,
};
