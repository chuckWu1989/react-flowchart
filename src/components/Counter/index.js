import React from 'react';
import { defaultProps, propTypes } from './props';
import './style.less';

class Counter extends React.Component {
  componentWillMount() {
    this.props.willMount();
  }
  componentWillUnmount() {
    this.props.onDispose();
  }
  render() {
    return (
      <div className="counter-style">
        <div className="count">
          <span>{this.props.count}</span>
        </div>
        <div className="buttons">
          <button onClick={this.props.increase}>Increase</button>
          <button onClick={this.props.decrease}>Decrease</button>
        </div>
      </div>
    );
  }
}
Counter.defaultProps = defaultProps;
Counter.propTypes = propTypes;

export default Counter;
