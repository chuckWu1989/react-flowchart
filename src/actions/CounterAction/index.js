import { createAction } from 'redux-actions';
import { SETCOUNT, DISPOSE } from '../../constants/ActionTypes';
import { defaultProps } from '../../components/Counter/props';

export const setCount = createAction(SETCOUNT);
export const dispose = createAction(DISPOSE);
export const decrease = props => (
  (dispatch, getState) => {
    const { id } = props;
    const state = getState();
    const count = state.getIn(['store', id, 'count']) - 1;
    dispatch(setCount({ id, count }));
  }
);
export const increase = props => (
  (dispatch, getState) => {
    const { id } = props;
    const state = getState();
    const count = state.getIn(['store', id, 'count']) + 1;
    dispatch(setCount({ id, count }));
  }
);
export const willMount = props => (
  (dispatch) => {
    const { id, count = defaultProps.count } = props;
    dispatch(setCount({ id, count }));
  }
);
export const onDispose = props => (
  (dispatch) => {
    const { id } = props;
    dispatch(dispose({ id }));
  }
);
