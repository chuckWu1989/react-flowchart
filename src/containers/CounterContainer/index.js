import { connect } from 'react-redux';
import Counter from '../../components/Counter';
import { willMount, increase, decrease, onDispose } from '../../actions/CounterAction';

export const mapStateToProps = (state, props) => ({
  count: state.getIn(['store', props.id, 'count']),
});
export const mapDispatchToProps = (dispatch, props) => ({
  willMount: () => dispatch(willMount(props)),
  increase: () => dispatch(increase(props)),
  decrease: () => dispatch(decrease(props)),
  onDispose: () => dispatch(onDispose(props)),
});
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Counter);
