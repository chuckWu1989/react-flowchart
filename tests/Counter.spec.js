import React from 'react';
import Immutable from 'immutable';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';
import Counter from '../src/components/Counter';
import { defaultProps } from '../src/components/Counter/props';
import { mapStateToProps, mapDispatchToProps } from '../src/containers/CounterContainer';
import * as actions from '../src/actions/CounterAction';
import * as actionTypes from '../src/constants/ActionTypes';

global.configure({ adapter: new Adapter() });

let store;
const props = { id: 'test' };
const mockStore = configureStore([thunk]);

describe('Counter', () => {
  describe('component', () => {
    it('should render correctly', () => {
      const ownProps = {
        ...props,
        count: 5,
        increase: response => response,
        decrease: response => response,
        willMount: response => response,
        onDispose: response => response,
      };
      const tree = toJson(global.shallow(<Counter {...ownProps} />));
      expect(tree).toMatchSnapshot();
    });
    it('should call handlers while component goes through lifecycle', () => {
      const ownProps = {
        ...props,
        willMount: jest.fn(),
        onDispose: jest.fn(),
      };
      const wrapper = global.shallow(
        <Counter {...ownProps} />,
        { lifecycleExperimental: true },
      );
      wrapper.unmount();
      expect(ownProps.willMount).toHaveBeenCalled();
      expect(ownProps.onDispose).toHaveBeenCalled();
    });
  });
  describe('container', () => {
    let mapState;
    let mapDispatch;
    beforeAll(() => {
      store = mockStore(Immutable.Map({}));
      mapState = mapStateToProps(store.getState(), props);
      mapDispatch = mapDispatchToProps(store.dispatch, props);
    });
    describe('mapStateToProps', () => {
      it('should pass count', () => {
        expect('count' in mapState).toBeTruthy();
      });
    });
    describe('mapDispatchToProps', () => {
      it('should pass willMount', () => {
        expect('willMount' in mapDispatch).toBeTruthy();
      });
      it('should pass increase', () => {
        expect('increase' in mapDispatch).toBeTruthy();
      });
      it('should pass decrease', () => {
        expect('decrease' in mapDispatch).toBeTruthy();
      });
      it('should pass onDispose', () => {
        expect('onDispose' in mapDispatch).toBeTruthy();
      });
    });
  });
  describe('action', () => {
    const count = 5;
    beforeEach(() => {
      store = mockStore(Immutable.Map({}).setIn(['store', props.id, 'count'], count));
    });
    describe('setCount', () => {
      it('should have properly action name', () => {
        const action = actions.setCount();
        expect(action.type).toBe(actionTypes.SETCOUNT);
      });
    });
    describe('dispose', () => {
      it('should have properly action name', () => {
        const action = actions.dispose();
        expect(action.type).toBe(actionTypes.DISPOSE);
      });
    });
    describe('willMount', () => {
      it('should set count into store if count is defined', () => {
        const ownProps = { ...props, count: 10 };
        store.dispatch(actions.willMount(ownProps));
        expect(store.getActions()[0].payload.count).toBe(10);
      });
      it('should set count as default if count is undefined', () => {
        store.dispatch(actions.willMount(props));
        expect(store.getActions()[0].payload.count).toBe(defaultProps.count);
      });
    });
    describe('increase', () => {
      it('should increase count', () => {
        store.dispatch(actions.increase(props));
        expect(store.getActions()[0].payload.count).toBe(count + 1);
      });
    });
    describe('decrease', () => {
      it('should decrease count', () => {
        store.dispatch(actions.decrease(props));
        expect(store.getActions()[0].payload.count).toBe(count - 1);
      });
    });
    describe('onDispose', () => {
      it('should relase data by id', () => {
        store.dispatch(actions.onDispose(props));
        expect(store.getActions()[0].payload.id).toBe(props.id);
      });
    });
  });
});
