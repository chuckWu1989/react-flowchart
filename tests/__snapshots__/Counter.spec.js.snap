// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should render correctly 1`] = `
<div
  className="counter-style"
>
  <div
    className="count"
  >
    <span>
      5
    </span>
  </div>
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
    >
      Increase
    </button>
    <button
      onClick={[Function]}
    >
      Decrease
    </button>
  </div>
</div>
`;
